### PR TITLE
test(flattable,select,actionpopover): add tests to check popover position top or bottom FE-3765

### PR DESCRIPTION
**Pull request by sianford**
### Proposed behaviour

Adding new Cypress tests for Action Popover, Select and Flat Table to assert the position of the popover is correct at above or below the input.

### Current behaviour

There are currently no Cypress tests to test this.

### Checklist

- [x] Commits follow our style guide
- [x] All themes are supported if required
- [x] Cypress automation tests added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
Run Cypress specs `cypress/features/regression/test/flatTableNoiFrame.feature`, `cypress/features/regression/test/simpleSelectNoIFrame.feature` and `cypress/features/regression/designSystem/actionPopover.feature`.

### CODE REVIEW

File: cypress/locators/flat-table/index.js [hunk 1]
```diff
@@ -2,8 +2,12 @@ import {
   FLAT_TABLE_COMPONENT,
   FLAT_TABLE_CELL,
   FLAT_TABLE_SUBROW,
+  FLAT_TABLE_PAGE_SIZE_SELECT,
+  FLAT_TABLE_PAGE_SELECT_LIST,
 } from "./locators";
 
+// import SELECT_LIST from "../select";
```
Comment (DlgSHi) [inline cypress/locators/flat-table/index.js:9]: please delete this if it is not needed any more

File: cypress/support/step-definitions/action-popover-steps.js [hunk 2]
```diff
@@ -78,3 +78,13 @@ Then("{string} action was called in Actions Tab for actionPopover", (event) => {
...
 });
+
+When("I scroll accordion content to top", () => {
+  cy.scrollTo("0", "0");
+});
+
+Then("Action Popover element is visible in {word} position", (position) => {
+  actionPopover()
+    .should("have.attr", "data-popper-placement", position)
+    .and("be.visible");
```
Comment (DlgSHi) [inline cypress/support/step-definitions/action-popover-steps.js:89]:
```suggestion
        .and("be.visible");
```

File: cypress/support/step-definitions/flat-table-steps.js [hunk 3]
```diff
@@ -244,3 +247,21 @@ Then("There is a shadow effect between the parent row and first subrow", () => {
...
+});
+
+When("pageSize is selected", () => {
+  flatTablePageSizeSelect().click();
+});
+
+Then("pageSizeSelectList is visible at the {word}", (position) => {
+  flatTablePageSelectListPosition()
+    .should("have.attr", "data-popper-placement", position)
+    .and("be.visible");
```
Comment (DlgSHi) [inline cypress/support/step-definitions/flat-table-steps.js:266]:
```suggestion
        .and("be.visible");
```

File: cypress/support/step-definitions/flat-table-steps.js [hunk 4]
```diff
@@ -244,3 +247,21 @@ Then("There is a shadow effect between the parent row and first subrow", () => {
...
+
+When("I have a large viewport", () => {
+  cy.viewport(700, 345);
+});
+
+When("I have a small viewport", () => {
+  cy.viewport(700, 240);
+});
+
+When("pageSize is selected", () => {
```
Comment (DlgSHi) [inline cypress/support/step-definitions/flat-table-steps.js:259]: we are performing `cy.click()` so it is not `selected` but `is opened` imo

File: cypress/features/regression/test/flatTableNoiFrame.feature [hunk 5]
```diff
@@ -1,14 +1,22 @@
 Feature: FlatTable component
-  I want to check FlatTable component properties
-
-  Background: Open FlatTable component page in no iFrame
-    Given I open Test default "Flat-Table" component in noIFrame with "flatTable" json from "test" using "default" object name
+  I want to check FlatTable component properties in no iFrame
 
   @positive
   Scenario: Header and row of FlatTable are visible after scrolling to the bottom right
+    Given I open Test default "Flat-Table" component in noIFrame with "flatTable" json from "test" using "default" object name
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/flatTableNoiFrame.feature:6]: I think it should also work :) , please check it 
```suggestion
    Given I open Test default "Flat Table" component in noIFrame with "flatTable" json from "test" using "default" object name
```

File: cypress/support/step-definitions/action-popover-steps.js [hunk 6]
```diff
@@ -78,3 +78,13 @@ Then("{string} action was called in Actions Tab for actionPopover", (event) => {
 Then("ActionPopover submenu is not visible", () => {
   actionPopoverSubmenuNoIFrame().should("not.be.visible");
 });
+
+When("I scroll accordion content to top", () => {
+  cy.scrollTo("0", "0");
+});
+
+Then("Action Popover element is visible in {word} position", (position) => {
+  actionPopover()
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/action-popover-steps.js:87]:
```suggestion
  actionPopover().should("have.attr", "data-popper-placement", position)
```

File: cypress/support/step-definitions/flat-table-steps.js [hunk 7]
```diff
@@ -244,3 +247,21 @@ Then("There is a shadow effect between the parent row and first subrow", () => {
...
+  flatTablePageSizeSelect().click();
+});
+
+Then("pageSizeSelectList is visible at the {word}", (position) => {
+  flatTablePageSelectListPosition()
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/flat-table-steps.js:264]:
```suggestion
  flatTablePageSelectListPosition().should("have.attr", "data-popper-placement", position)
```

File: cypress/support/step-definitions/flat-table-steps.js (hunk 3 above)
Comment (sianford) [reply cypress/support/step-definitions/flat-table-steps.js:266]: As discussed, I left the format as it is to keep it uniform with the other steps in the file.

File: cypress/support/step-definitions/flat-table-steps.js (hunk 4 above)
Comment (sianford) [reply cypress/support/step-definitions/flat-table-steps.js]: Amended the step in the feature file and in the step def to make it more readable.

File: cypress/support/step-definitions/flat-table-steps.js (hunk 7 above)
Comment (sianford) [reply cypress/support/step-definitions/flat-table-steps.js:264]: As discussed, I left the format as it is to keep it uniform with the other steps in the file. I logged a ticket to standardise all step definition files as their formats and indents differ.

File: cypress/features/regression/test/flatTableNoiFrame.feature [hunk 8]
```diff
@@ -1,14 +1,22 @@
...
       And First 2 header cells aren't visible
       And Last 4 FlatTable rows are visible
       And First 3 FlatTable rows aren't visible
 
+@positive
+  Scenario Outline: Show page size select list of FlatTable is at the <position> in <size> viewport
+    Given I open "Design System Flat Table" component page "paginated with sticky header" in no iframe
+      And I have a <size> viewport
+    When pageSize select list is opened
+    Then pageSizeSelectList is visible at the <position>
```
Comment (DlgSHi) [inline cypress/features/regression/test/flatTableNoiFrame.feature:18]: could we move `-start` as ot used in `top` and `bottom` positions into step

```
pageSizeSelectList is visible at the

File: cypress/features/regression/test/simpleSelectNoIFrame.feature [hunk 9]
```diff
@@ -0,0 +1,13 @@
+Feature: Select component
+  I want to check SimpleSelect component list position in no iFrame
+  
+  @positive
+    Scenario Outline: Show Select list is at the <position> in <size> viewport
+      Given I open Test default "Select" component in noIFrame with "simpleSelect" json from "test" using "<size>" object name
+        And I have a <size> viewport
+      When I click on Select input in noIframe
+      Then "simple" Select list is visible at the <position>
```
Comment (DlgSHi) [inline cypress/features/regression/test/simpleSelectNoIFrame.feature:9]: the same is here, move please `-end` into the step

File: cypress/fixtures/test/simpleSelect.json [hunk 10]
```diff
@@ -0,0 +1,8 @@
+{ 
+    "large": {
+      "mt": "0"
```
Comment (DlgSHi) [inline cypress/fixtures/test/simpleSelect.json:3]: what is the purpose of passing an empty prop?

File: cypress/support/step-definitions/action-popover-steps.js [hunk 11]
```diff
@@ -78,3 +78,12 @@ Then("{string} action was called in Actions Tab for actionPopover", (event) => {
 Then("ActionPopover submenu is not visible", () => {
   actionPopoverSubmenuNoIFrame().should("not.be.visible");
 });
+
+When("I scroll accordion content to top", () => {
+  cy.scrollTo("0", "0");
+});
+
+Then("Action Popover element is visible in {word} position", (position) => {
+  actionPopover().should("have.attr", "data-popper-placement", position)
```
Comment (DlgSHi) [inline cypress/support/step-definitions/action-popover-steps.js:87]:
```suggestion
  actionPopover().should("have.attr", "data-popper-placement", `${position}-end`)
```

File: cypress/support/step-definitions/flat-table-steps.js [hunk 12]
```diff
@@ -244,3 +247,21 @@ Then("There is a shadow effect between the parent row and first subrow", () => {
...
+  cy.viewport(700, 240);
+});
+
+When("pageSize select list is opened", () => {
+  flatTablePageSizeSelect().click();
+});
+
+Then("pageSizeSelectList is visible at the {word}", (position) => {
+  flatTablePageSelectListPosition()
+    .should("have.attr", "data-popper-placement", position)
```
Comment (DlgSHi) [inline cypress/support/step-definitions/flat-table-steps.js:265]:
```suggestion
    .should("have.attr", "data-popper-placement", `${position}-start)
```

File: cypress/support/step-definitions/select-steps.js [hunk 13]
```diff
@@ -225,6 +226,12 @@ Then("Select input has no value", () => {
   getDataElementByValue("input").should("have.attr", "value", "");
 });
 
+Then("{string} Select list is visible at the {word}", (name, position) => {
+  selectListPosition(name)
+    .should("have.attr", "data-popper-placement", position)
```
Comment (DlgSHi) [inline cypress/support/step-definitions/select-steps.js:231]: and the same is here:
```suggestion
    .should("have.attr", "data-popper-placement", `${position}-end`)
```

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: cypress/support/step-definitions/action-popover-steps.js (hunk 11 above)
Comment (sianford) [reply cypress/support/step-definitions/action-popover-steps.js:87]: Good suggestions, thanks 👍

**Review by DlgSHi — DISMISSED** [review]

**Review by KatarzynaQA — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**